x/ibc/applications/transfer/types/keys.go

```diff
@@ -39,11 +43,15 @@ var (
 	DenomTraceKey = []byte{0x02}
 )
 
-// GetEscrowAddress returns the escrow address for the specified channel
+// GetEscrowAddress returns the escrow address for the specified channel.
+// The escrow address uses a prefix to prevent address collsions with Public Key addresses.
 //
 // CONTRACT: this assumes that there's only one bank bridge module that owns the
 // port associated with the channel ID so that the address created is actually
 // unique.
 func GetEscrowAddress(portID, channelID string) sdk.AccAddress {
-	return sdk.AccAddress(crypto.AddressHash([]byte(fmt.Sprintf("%s/%s", portID, channelID))))
+	return append(
+		[]byte(EscrowPrefix),
```

Comment:
I think this creates an invalid account address as the length will be larger. 
What about 

```go
crypto.AddressHash(
	append(
		crypto.AddressHash([]byte(EscrowPrefix)),
		crypto.AddressHash([]byte(fmt.Sprintf("%s/%s", portID, channelID)))...,
	)
)
```

Reply:
I think the purpose is to ensure the address length isn't 20 bytes so it doesn't collide with other addresses

Reply:
From my understanding the address length is only used with Bech32 addresses which doesn't seem to be an enforcement to be a valid sdk.AccAddress

Reply:
tests are also passing so if `types.AddrLen` is supposed to be enforced, it is not for calling sends directly or minting coins

Reply:
yes, the intention is specifically to create something longer than an address

Reply:
but then it's not an address anymore